editor/components/provider/index.js

```diff
@@ -20,7 +20,7 @@ import {
  * Internal Dependencies
  */
 import { setupEditor, undo } from '../../actions';
-import createReduxStore from '../../store';
+import store from '../../store';
```

Comment:
It's a bit unnerving to me that we treat store as a singleton instance, since there's nothing stopping someone from importing from store to get state or dispatch from anywhere in the codebase. This makes it a little less predictable how state is being interacted with. In this way, it's more similar to the old Flux dispatcher approach. One of the reasons we moved away from Flux to Redux in Calypso was that sharing the store as a singleton could [lead to session data leaking between server-side requests](https://github.com/Automattic/wp-calypso/blob/master/docs/our-approach-to-data.md#third-era-redux-global-state-tree-december-2015---present). Obviously this doesn't apply to the purely client-side editor, so it may not be as big a concern.

Reply:
The main idea here is that a WP module requiring state should register itself as soon as the script is loaded (other plugins might depend on this module, thus, on its state).

A module = 1 state tree (or 0).

We shouldn't wait for the provider to be mounted to load initialize the state of the module.

So this PR is changing the meaning of the "editor" state from editor instance's state to editor module's state which I think is the way to go to "modularize" the state in a centralized "data" module. 

Later, we can decide to add a root level to the "editor" module's reducer containting the "id" of the instanciated editor (which would mean we could hold multiple editor instances)